src/octoprint/plugins/softwareupdate/__init__.py

```diff
@@ -712,6 +713,16 @@ def get_settings_defaults(self):
     def on_settings_load(self):
         # ensure we don't persist check configs we receive on the API
         data = dict(octoprint.plugin.SettingsPlugin.on_settings_load(self))
+
+        # set credentials data flags and remove credentials fields
+        if "credentials" in data:
+            if "github" in data["credentials"]:
+                data["credentials"]["github_set"] = bool(data["credentials"]["github"])
+            for key in data["credentials"]:
+                if key == "github_set":
+                    continue
+                data["credentials"][key] = None
+
```

Comment:
Instead of all of this which basically duplicates logic that is already handled by `get_settings_restricted_paths`, all just because the flag is put into the `credentials` field on the default settings (see above), how about this:

```suggestion

        # set credentials flag
        data["credentials"] = {"github_set": bool(self._settings.get(["credentials", "github"])) }
```

Or, even better (because it will automatically adapt to more credentials added in the future):

```suggestion

        # set credentials flag
        credentials = self._settings.get(["credentials"])
        data["credentials"] = {}
        for key, value in credentials.items():
            data["credentials"][f"{key}_set"] = value is not None
```

Reply:
Thanks for the explanation.
I thought that from the plugin it was only possible to access the settings allowed by `get_settings_restricted_paths`